lib/components/mobile/batch-results-screen.js

```diff
@@ -0,0 +1,142 @@
+import coreUtils from '@opentripplanner/core-utils'
+import React, { Component } from 'react'
+import { Button } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+
+import * as uiActions from '../../actions/ui'
+import Map from '../map/map'
+import Icon from '../narrative/icon'
+import NarrativeItineraries from '../narrative/narrative-itineraries'
+import {
+  getActiveItineraries,
+  getActiveSearch,
+  getResponsesWithErrors
+} from '../../util/state'
+
+import MobileContainer from './container'
+import ResultsError from './results-error'
+import ResultsHeader from './results-header'
+
+const StyledMobileContainer = styled(MobileContainer)`
+  .options > .header {
+    margin: 10px;
+  }
+
+  &.otp.mobile .mobile-narrative-container {
+    bottom: 0;
+    left: 0;
+    overflow-y: auto;
+    padding: 0;
+    position: fixed;
+    right: 0;
+  }
+`
+
+const ExpandMapButton = styled(Button)`
+  bottom: 10px;
+  left: 10px;
+  position: absolute;
+  z-index: 999999;
+`
+
+const VIEW = uiActions.ItineraryView
+
+const NARRATIVE_SPLIT_TOP_PERCENT = 45
+
+/**
+ * This component renders the mobile view of itinerary results from batch routing,
+ * and features a split view between the map and itinerary results or narratives.
+ */
+class BatchMobileResultsScreen extends Component {
+  componentDidUpdate (prevProps) {
+    if (this.props.activeLeg !== prevProps.activeLeg) {
+      // Check if the active leg has changed. If a different leg is selected,
+      // unexpand the itinerary to show the map focused on the selected leg
+      // (similar to the behavior of LineItinerary).
+      this._setItineraryExpanded(false)
+    }
+  }
+
+  _setItineraryExpanded = itineraryExpanded => {
+    this.props.setItineraryView(itineraryExpanded ? VIEW.FULL : VIEW.SPLIT)
+  }
+
+  _toggleMapExpanded = () => {
+    this.props.setItineraryView(this.props.itineraryView === VIEW.HIDDEN ? VIEW.SPLIT : VIEW.HIDDEN)
+  }
+
+  render () {
+    const { errors, itineraries, itineraryView } = this.props
+    const hasNoResult = itineraries.length === 0 && errors.length > 0
+    const mapExpanded = itineraryView === VIEW.HIDDEN
+    const itineraryExpanded = itineraryView === VIEW.FULL
+    const narrativeTop = mapExpanded ? '100%' : (itineraryExpanded ? '100px' : `${NARRATIVE_SPLIT_TOP_PERCENT}%`)
+    const mapBottom = mapExpanded ? 0 : `${100 - NARRATIVE_SPLIT_TOP_PERCENT}%`
+
+    return (
+      <StyledMobileContainer>
+        <ResultsHeader />
+        <div
+          className='results-map'
+          style={{bottom: mapBottom, display: itineraryExpanded ? 'none' : 'inherit'}}
```

Comment:
This could be part of a styled component that accepts props and styles accordingly.

Reply:
Generally, existing CSS doesn't play nice with styled components. Making this change involves plucking the existing CSS apart, and I would stay away from that until further discussion.

Reply:
Alright, this is converted in https://github.com/opentripplanner/otp-react-redux/pull/339/commits/d7737831bffd7e6a4f5294541b0187c1a2574c3a.